app/services/rapid_ready_for_decision/hypertension_upload_manager.rb

```diff
@@ -4,15 +4,24 @@ module RapidReadyForDecision
   class HypertensionUploadManager
     attr_accessor :submission
 
+    DOCUMENT_TITLE = 'VAMC_Hypertension_Rapid_Decision_Evidence'
+
     def initialize(submission)
       @submission = submission
     end
 
+    def file_upload_name
+      end_date = Time.zone.today.strftime('%m%d%Y')
+      start_date = 1.year.ago.strftime('%m%d%Y')
+
+      @file_upload_name ||= "#{DOCUMENT_TITLE}--#{start_date}-#{end_date}.pdf"
```

Comment:
```suggestion
      @file_upload_name ||= begin
        end_date = Time.zone.today.strftime('%m%d%Y')
        start_date = 1.year.ago.strftime('%m%d%Y')
        "#{DOCUMENT_TITLE}--#{start_date}-#{end_date}.pdf"
      end
```

Reply:
@yoomlam very cool, thanks — didn't know about this Ruby feature. What is the difference as far as behavior, though? Or is it just clearer? My understanding is that `@file_upload_name ||= "#{DOCUMENT_TITLE}--#{start_date}-#{end_date}.pdf"` would still use the memoized value, although it's true that the datetime values would still have been recomputed on subsequent calls.... thanks!

Reply:
> although it's true that the datetime values would still have been recomputed on subsequent calls

Yea, moving that code in the `begin` block saves a few cpu cycles and needless string computation. Search for `||= begin` in other parts of the code for other examples.